src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HealthHeartbeatDefaultPayload.cs

```diff
@@ -0,0 +1,125 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using System.Reflection;
+
+    internal class HealthHeartbeatDefaultPayload
+    {
+        public const string UpdatedFieldsPropertyKey = "updatedFields";
+
+        public static readonly string[] DefaultFields =
+        {
+            "runtimeFramework",
+            "targetFramework",
+            UpdatedFieldsPropertyKey
+        };
+
+        private List<string> enabledProperties;
+
+        public HealthHeartbeatDefaultPayload() : this(null)
+        {
+        }
+
+        public HealthHeartbeatDefaultPayload(IEnumerable<string> disableFields)
+        {
+            this.SetEnabledProperties(disableFields);
+        }
+
+        public IDictionary<string, HealthHeartbeatPropertyPayload> GetPayloadProperties()
+        {
+            var payload = new Dictionary<string, HealthHeartbeatPropertyPayload>();
+            foreach (string fieldName in this.enabledProperties)
+            {
+                switch (fieldName)
+                {
+                    case "runtimeFramework":
+                        payload.Add(fieldName, new HealthHeartbeatPropertyPayload()
+                        {
+                            IsHealthy = true,
+                            PayloadValue = this.GetRuntimeFrameworkVer()
+                        });
+                        break;
+                    case "targetFramework":
```

Comment:
as commented before - this is not actually a target framework. It is a base SDK target framework. I'd suggest to make it explicit in the name of the metadata property.